src/autosuggest/options-controller.ts

```diff
@@ -101,7 +101,9 @@ function createItems(options: Options): AutosuggestItem[] {
   const items: AutosuggestItem[] = [];
   for (const option of options) {
     if (isGroup(option)) {
-      items.push(...flattenGroup(option));
```

Comment:
That line caused an exception when too many options. See [MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply): 

"But beware: by using apply this way, you run the risk of exceeding the JavaScript engine's argument length limit. The consequences of applying a function with too many arguments (that is, more than tens of thousands of arguments) varies across engines. (The JavaScriptCore engine has hard-coded argument limit of 65536.)"

An exception inside use-memo triggered a React.js exception saying an infinite update loop was detected, which is wrong.

Reply:
1. This is not very obvious, how about leaving a comment saying why `.push(...arr)` does not work?
2. This does not seem like a root cause of the originally reported error (react rendered too many items). I am guessing we prevented creating copies of objects, but I do not see it anywhere. Could you help me to understand the core of the change here?

Reply:
Try this:

```
let items = []
for (let i = 0; i < 200_000; i++) {
  items.push(i)
}
[].push(...items) // Uncaught RangeError: Maximum call stack size exceeded
```

Reply:
The uncaught exception inside useMemo callback triggers a React exception in turn

Reply:
Good, I did not know that! One more reason to avoid using memos without reasons, because they swallow the real exceptions.

Reply:
I guess here we have a very clear performance gain from it, but need to be careful indeed.